tools/test-proxy/Azure.Sdk.Tools.TestProxy/RecordingHandler.cs

```diff
@@ -263,21 +261,27 @@ public async Task StartPlayback(string sessionId, HttpResponse outgoingResponse,
             {
                 if (!InMemorySessions.TryGetValue(sessionId, out session))
                 {
-                    throw new InvalidOperationException("Failed to retrieve in-memory session.");
+                    throw new HttpException(HttpStatusCode.BadRequest, $"There is no in-memory session with id {id} available for playback retrieval.");
                 }
                 session.SourceRecordingId = sessionId;
             }
             else
             {
-                using var stream = System.IO.File.OpenRead(GetRecordingPath(sessionId));
+                var path = GetRecordingPath(sessionId);
+
+                if (!File.Exists(path))
+                {
+                    throw new HttpException(HttpStatusCode.BadRequest, $"Recording file path {path} does not exist.");
```

Comment:
Should this be a TestRecordingMismatchException?

Reply:
The way I think about it currently? No, but it would be easy to change.

My perspective, `badrequest` any time it's a user error. Starting playback against a recording that does not exist is a user error in my mind. It's not a `mismatchexception` because there wasn't any matching taking place at all!

That being said, I can see the argument from the perspective of 

> failing to load a recording IS a mismatch

But they're different in my brain. One is a failure to record before playback, the other is an unexpected failure during playback.

Reply:
lmk what you think about the above, because i'm adding tests and have a bit of time :P

Reply:
Weird I somehow managed to edit your comment instead of replying.

Reply:
> But they're different in my brain. One is a failure to record before playback, the other is an unexpected failure during playback.


We treat them the same in .NET framework because we have a feature that allows automatic recording of tests that fail playback. So when a test is first added, it would be run in playback but then should be automatically recorded. I think it is fine to leave this out and have it be custom behavior in .NET.

Reply:
That's fair! Then let's swap this one to a TestMismatchException here then 👍

Reply:
Ah, I misread your original point here. Have you already added handling code for this?

Reply:
Yeah either way is fine - it would make things a bit more consistent to use TestMismatchException provided that also will automatically set the x-request-mismatch header.

Reply:
Yep. It'll definitely be set! I've updated this loc.